cql3/restrictions/statement_restrictions.cc

```diff
@@ -916,6 +915,18 @@ bool token_restriction::slice::is_satisfied_by(const schema& schema,
     return satisfied;
 }
 
+bool single_column_restriction::LIKE::init_matcher(const query_options& options) const {
+    auto pattern = to_bytes_opt(_value->bind_and_get(options));
+    if (!pattern) {
+        return false;
+    }
+    if (!_matcher || pattern != _last_pattern) {
+        _matcher.reset(new like_matcher(*pattern));
+        _last_pattern = std::move(pattern);
```

Comment:
std::make_unique<> is better practice than a naked new. std::optional<> would be even better.